addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,115 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+from requests.exceptions import HTTPError, Timeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+    :raises UserError: If there isn't any product without a picture and with a barcode selected.
+    """
+    _name = 'product.images.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields):
+        default = super().default_get(fields)
+
+        google_custom_search_cx = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.cx'))
+        google_custom_search_key = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.key'))
+
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(self._context.get('active_ids')) \
+                .product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(self._context.get('active_ids')) \
+                .filtered(lambda p: not p.image_1920 and p.barcode).ids
+
+        if len(product_ids) == 0:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+        # TODO: Define a maximum number of product to process in the same batch
+        if len(product_ids) > 100:
+            raise UserError(_("You selected too many product variants at the same time. \
+Please select a maximum of 100 product variants in the same batch."))
+
+        default.update({
+            'product_ids': product_ids,
+            'nb_product': len(product_ids),
+        })
+
+        return default
+
+    product_ids = fields.Many2many(
+        comodel_name="product.product",
+        help="List of the selected product without a picture and with a barcode",
+    )
+    nb_product = fields.Integer(string="Number of product selected", readonly=True)
+
+    def get_pic_using_barcode(self):
+        for record in self:
+            for product in record.product_ids:
+                product.image_1920 = self._get_pic_from_google(product.barcode)
```

Comment:
Given `_get_pic_from_google` is making at least two calls to "https://customsearch.googleapis.com" per input product (up to 100), and possibly repeated calls to a third-party if all the search yield similar hits (e.g. we're adding images to products from a single supplier so all the hits are to them) this should really use a `request.Session` to leverage persistent connections.

Sadly session-level timeout support [while feasible](https://github.com/psf/requests/issues/3341#issuecomment-226763313) still is not supported out of the box, so that wouldn't really obviate the need for per-request timeouts.